sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs

```diff
@@ -394,9 +420,20 @@ public virtual async Task<Response> UploadAsync<T>(string ruleId, string streamN
                 else
                 {
                     Exception exceptionEventHandler = await ProcessCompletedTaskEventHandlerAsync(task.CurrentTask, task.Logs, options, cancellationToken).ConfigureAwait(false);
-                    shouldAbort = exceptionEventHandler != null;
-                    if (shouldAbort)
+                    logsFailed += task.Logs.Count;
+                    // if exception is thrown stop processing future batches
+                    if (exceptionEventHandler != null)
+                    {
+                        shouldAbort = true;
                         AddException(ref exceptions, exceptionEventHandler);
+                    }
+                    // Cancel all future Uploads if user triggers CancellationToken
+                    // if exception was not thrown on line 400 - token could still be cancelled
+                    if (cancellationToken.IsCancellationRequested)
```

Comment:
If no OperationCanceledException is thrown, then it means no downstream work actually got canceled despite the user's attempt to cancel. So there is no need to throw an exception ourselves for this case.